go/vt/dbconnpool/connection_pool.go

```diff
@@ -139,6 +204,17 @@ func (cp *ConnectionPool) Get(ctx context.Context) (*PooledDBConnection, error)
 	if err != nil {
 		return nil, err
 	}
+
+	// Check that the RemoteAddr is still a valid Address
+	if cp.resolutionFrequency > 0 &&
+		net.ParseIP(cp.info.Host) == nil &&
```

Comment:
The `cp.info.Host` shouldn't change for the lifetime of the pool, right? Can we store a bool in `cp` so we don't have to call `ParseIP` every time?